src/identity-provider.js

```diff
@@ -1,57 +1,23 @@
 'use strict'
-const isDefined = require('./is-defined')
-const Identity = require('./identity')
-const defaultType = 'orbitdb'
-
 class IdentityProvider {
-  constructor (keystore) {
-    if (!isDefined(keystore)) {
-      throw new Error('Keystore is required')
-    }
-    this._keystore = keystore
-  }
-
-  async createIdentity(id, options = {}) {
-    // Get the key for id from the keystore or create one
-    // if it doesn't exist
-    const key = await this._keystore.getKey(id) ||
-      await this._keystore.createKey(id)
-    // Sign with the key for the id
-    const selfSigningFn = async (id, data) => await this._keystore.sign(key, data)
-    // If no type was indicated, set to default
-    const type = isDefined(options.type) ? options.type : defaultType
-    // If signing function was not passed, use keystore as the identity signer
-    const identitySignerFn = isDefined(options.identitySignerFn) ? options.identitySignerFn : selfSigningFn
-    // Sign the id with the signing key we're going to use
-    const idSignature = await this._keystore.sign(key, id)
-    // Get the hex string of the public key
-    const publicKey = key.getPublic('hex')
-    // Sign both the key and the signature created with that key
-    const pubKeyIdSignature = await identitySignerFn(id, publicKey + idSignature)
-    return new Identity(id, publicKey, idSignature, pubKeyIdSignature, type, this)
-  }
+  constructor() {}
 
-  static async createIdentity (keystore, id, options = {}) {
-    const identityProvider = new IdentityProvider(keystore)
-    return await identityProvider.createIdentity(id, options)
-  }
-
-  static async verifyIdentity (identity, verifierFunction) {
-    return verifierFunction(identity)
-  }
+  /* Return publicKey of identity */
+  async createId(options) {}
```

Comment:
I think this should be called `getId(...)` instead of `createId`. We're not "creating a new id" but returning the id based on the inputs (args) to the IdentityProvider, right?

Reply:
Yea its true and also quite confusing to call it xxxxId()... for verification this should be the external publicKey used to sign the odb pubKey so I think "getPublicKey" is better, hows that?

Reply:
We have `publicKey` separately in the Identity which is the key used to sign with this identity. What you're referring to as "publicKey" in the previous comment is used as the *id* of the Identity, and it can be something else than a public key (eg. assigned by a centralized service). I think it's important to separate the two conceptually and avoid the confusion with rest of the Identity data structure.